test/com/ms/silverking/cloud/dht/client/impl/AsyncRetrievalOperationImplTest.java

```diff
@@ -0,0 +1,84 @@
+package com.ms.silverking.cloud.dht.client.impl;
+
+import static org.mockito.Mockito.doCallRealMethod;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import com.ms.silverking.cloud.dht.client.FailureCause;
+import com.ms.silverking.cloud.dht.client.KeyedOperationException;
+import com.ms.silverking.cloud.dht.client.OperationException;
+import com.ms.silverking.cloud.dht.client.OperationState;
+import com.ms.silverking.cloud.dht.client.StoredValue;
+import com.ms.silverking.thread.lwt.BaseWorker;
+import com.ms.silverking.thread.lwt.LWTPool;
+import com.ms.silverking.thread.lwt.LWTPoolProvider;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+public class AsyncRetrievalOperationImplTest {
+
+    AsyncRetrievalOperationImpl<String, String> retrievalOperationMock;
+
+    @Before
+    public void init() {
+        BaseWorker.class.getClassLoader().setClassAssertionStatus(BaseWorker.class.getName(), false);
+        LWTPoolProvider.defaultNonConcurrentWorkPool = mock(LWTPool.class);
+
+        retrievalOperationMock = (AsyncRetrievalOperationImpl<String, String>) mock(AsyncRetrievalOperationImpl.class);
+        when(retrievalOperationMock.getOperationStateMap()).thenReturn(new HashMap<String, OperationState>());
+        when(retrievalOperationMock.getPartialResults()).thenReturn(new HashMap<String, StoredValue>());
+    }
+
+    @After
+    public void finalize() {
```

Comment:
Our static analysis engine detects that your pull request includes a change with a **public finalizer**. According to [the best practice](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#fi-finalizer-should-be-protected-not-public-fi-public-should-be-protected), a class's `finalize()` method should have protected access, not public